# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build libassuan-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.71; am-1.16; lt-2.4.2)
libtoolize -c -f; aclocal; autoheader; autoconf; automake -a -c
./configure --disable-gpg-test --disable-languages --prefix=

# build
make

# install
install -d $DESTDIR/bin
install src/.libs/gpgme-json $DESTDIR/bin
install -d $DESTDIR/include
install -m 644 src/gpgme.h $DESTDIR/include
install -d $DESTDIR/lib
install src/.libs/libgpgme.so.11 $DESTDIR/lib
ln -sf libgpgme.so.11 $DESTDIR/lib/libgpgme.so
install -d $DESTDIR/lib/pkgconfig
install -m 644 src/gpgme.pc $DESTDIR/lib/pkgconfig
install -d $DESTDIR/share/aclocal
install -m 644 src/gpgme.m4 $DESTDIR/share/aclocal

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/gpgme-build.pc << EOF
Name: gpgme
Version: 1.23.2
Description: GnuPG Made Easy library
Requires: $REQUIRES

devel=\\
/include/gpgme.h \\
/lib/libgpgme.so \\
/lib/pkgconfig/gpgme.pc \\
/share/aclocal/gpgme.m4

exec=\\
/bin/gpgme-json \\
/lib/libgpgme.so.11
EOF

exit

# patch
--- gpgme-1.23.2.orig/m4/gpg-error.m4
+++ gpgme-1.23.2/m4/gpg-error.m4
@@ -126,16 +126,10 @@
 
   if test -n "$gpgrt_libdir"; then
     GPGRT_CONFIG="$GPGRT_CONFIG --libdir=$gpgrt_libdir"
-    if $GPGRT_CONFIG gpg-error >/dev/null 2>&1; then
-      GPG_ERROR_CONFIG="$GPGRT_CONFIG gpg-error"
-      AC_MSG_NOTICE([Use gpgrt-config with $gpgrt_libdir as gpg-error-config])
-      gpg_error_config_version=`$GPG_ERROR_CONFIG --modversion`
-    else
-      gpg_error_config_version=`$GPG_ERROR_CONFIG --version`
+    if ! $GPGRT_CONFIG gpg-error --exists; then
       unset GPGRT_CONFIG
     fi
-  elif test "$GPG_ERROR_CONFIG" != "no"; then
-    gpg_error_config_version=`$GPG_ERROR_CONFIG --version`
+  else
     unset GPGRT_CONFIG
   fi
 ])
@@ -145,7 +139,7 @@
 dnl
 dnl Test for libgpg-error and define GPG_ERROR_CFLAGS, GPG_ERROR_LIBS,
 dnl GPG_ERROR_MT_CFLAGS, and GPG_ERROR_MT_LIBS.  The _MT_ variants are
-dnl used for programs requireing real multi thread support.
+dnl used for programs requiring real multi thread support.
 dnl
 dnl If a prefix option is not used, the config script is first
 dnl searched in $SYSROOT/bin and then along $PATH.  If the used
@@ -156,6 +150,23 @@
 AC_REQUIRE([AC_CANONICAL_HOST])dnl
 AC_REQUIRE([_AM_PATH_POSSIBLE_GPG_ERROR_CONFIG])dnl
 AC_REQUIRE([_AM_PATH_GPGRT_CONFIG])dnl
+  if test x"$GPGRT_CONFIG" != x -a "$GPGRT_CONFIG" != "no"; then
+    GPG_ERROR_CONFIG="$GPGRT_CONFIG gpg-error"
+    AC_MSG_NOTICE([Use gpgrt-config with $gpgrt_libdir as gpg-error-config])
+    gpg_error_config_version=`$GPG_ERROR_CONFIG --modversion`
+  elif test x"$GPG_ERROR_CONFIG" != x -a "$GPG_ERROR_CONFIG" != "no"; then
+    gpg_error_config_version=`$GPG_ERROR_CONFIG --version`
+  else
+    if test "x$ac_cv_env_PKG_CONFIG_set" != "xset"; then
+      AC_PATH_TOOL([PKG_CONFIG], [pkg-config])
+    fi
+    GPG_ERROR_CONFIG="$PKG_CONFIG gpg-error"
+    if `$GPG_ERROR_CONFIG --exists`; then
+      gpg_error_config_version=`$GPG_ERROR_CONFIG --modversion`
+    else
+      GPG_ERROR_CONFIG=no
+    fi
+  fi
   min_gpg_error_version=ifelse([$1], ,1.33,$1)
   ok=no
   if test "$GPG_ERROR_CONFIG" != "no"; then
--- gpgme-1.23.2.orig/m4/libassuan.m4
+++ gpgme-1.23.2/m4/libassuan.m4
@@ -38,6 +38,17 @@
   if test -z "$use_gpgrt_config"; then
     AC_PATH_PROG(LIBASSUAN_CONFIG, libassuan-config, no)
   fi
+  if test "$LIBASSUAN_CONFIG" = "no"; then
+    if test "x$ac_cv_env_PKG_CONFIG_set" != "xset"; then
+      AC_PATH_TOOL([PKG_CONFIG], [pkg-config])
+    fi
+    LIBASSUAN_CONFIG="$PKG_CONFIG libassuan"
+    if `$LIBASSUAN_CONFIG --exists`; then
+      use_gpgrt_config=yes
+    else
+      LIBASSUAN_CONFIG=no
+    fi
+  fi
 
   tmp=ifelse([$1], ,1:0.9.2,$1)
   if echo "$tmp" | grep ':' >/dev/null 2>/dev/null ; then
--- gpgme-1.23.2.orig/src/gpgme.m4
+++ gpgme-1.23.2/src/gpgme.m4
@@ -87,15 +87,10 @@
 
   if test -n "$gpgrt_libdir"; then
     GPGRT_CONFIG="$GPGRT_CONFIG --libdir=$gpgrt_libdir"
-    if $GPGRT_CONFIG gpg-error >/dev/null 2>&1; then
-      GPG_ERROR_CONFIG="$GPGRT_CONFIG gpg-error"
-      AC_MSG_NOTICE([Use gpgrt-config with $gpgrt_libdir as gpg-error-config])
-      gpg_error_config_version=`$GPG_ERROR_CONFIG --modversion`
-    else
+    if ! $GPGRT_CONFIG gpg-error --exists; then
       unset GPGRT_CONFIG
     fi
   elif test "$GPG_ERROR_CONFIG" != "no"; then
-    gpg_error_config_version=`$GPG_ERROR_CONFIG --version`
     unset GPGRT_CONFIG
   fi
 ])
@@ -136,6 +131,17 @@
   if test -z "$use_gpgrt_config"; then
     AC_PATH_PROG(GPGME_CONFIG, gpgme-config, no)
   fi
+  if test "$GPGME_CONFIG" = "no"; then
+    if test "x$ac_cv_env_PKG_CONFIG_set" != "xset"; then
+      AC_PATH_TOOL([PKG_CONFIG], [pkg-config])
+    fi
+    GPGME_CONFIG="$PKG_CONFIG gpgme"
+    if `$GPGME_CONFIG --exists`; then
+      use_gpgrt_config=yes
+    else
+      GPGME_CONFIG=no
+    fi
+  fi
 
   if test "$GPGME_CONFIG" != "no" ; then
     if test -z "$use_gpgrt_config"; then

# source
https://gnupg.org/ftp/gcrypt/gpgme/gpgme-1.23.2.tar.bz2
